Pre-DevIncubator/Vehicle.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Pre_DevIncubator
+{
+    public class Vehicle : IComparable<Vehicle>
+    {
+        private readonly VehicleType vehicleType;
+        private readonly string modelName;
+        private string registrationNumber;
+        private int weight;
+        private readonly int manufactureYear;
+        private int mileage;
+        private Color color;
+
+        public VehicleType VehicleType => vehicleType;
+
+        public string ModelName => modelName;
+
+        public string RegistrationNumber { get => registrationNumber; set => registrationNumber = value; }
+        public int Weight { get => weight; set => weight = value; }
+
+        public int ManufactureYear => manufactureYear;
+
+        public int Mileage { get => mileage; set => mileage = value; }
+        public Color Color { get => color; set => color = value; }
+
+        public Vehicle()
+        {
+            vehicleType = new VehicleType();
+            modelName = string.Empty;
+            registrationNumber = string.Empty;
+            weight = 0;
+            manufactureYear = 0;
+            mileage = 0;
+            color = Color.Blue;
```

Comment:
This constructor looks quite strange. You can get rid of initializing the fields in this constructor by initializing them in the class definition.
```suggestion
//Smt like this
        public VehicleType VehicleType {get; init;} = new();
        public string ModelName {get; init;} = "defaultModel";
        public string RegistrationNumber { get; set; }
        public int Weight { get; set; }
        public int ManufactureYear {get; init;}
        public int Mileage { get; set; }
        public Color Color { get; set;} = Color.Blue;
        
        public Vehicle()
        {
        }
```